File: attached_assets/Starter_code/Flutter_starter_code/lib/src/models/policy.dart
class Policy {
  final String id;
  final String policyNumber;
  final String type;
  final String customerName;
  final double premium;
  final double coverage;
  final String startDate;
  final String endDate;
  final String status;

  Policy({required this.id, required this.policyNumber, required this.type, required this.customerName, required this.premium, required this.coverage, required this.startDate, required this.endDate, required this.status});

  factory Policy.fromJson(Map<String, dynamic> j) => Policy(
    id: (j['id'] ?? j['policyNumber']).toString(),
    policyNumber: j['policyNumber'],
    type: j['type'],
    customerName: j['customerName'],
    premium: (j['premium'] as num).toDouble(),
    coverage: (j['coverage'] as num).toDouble(),
    startDate: j['startDate'],
    endDate: j['endDate'],
    status: j['status'],
  );
}

File: attached_assets/Starter_code/Java_starter_code/src/main/java/com/example/insurance/domain/Policy.java
package com.example.insurance.domain;

import jakarta.persistence.*;
import java.time.LocalDate;
import java.util.*;

@Entity
public class Policy {
    @Id @GeneratedValue(strategy = GenerationType.UUID)
    private String id;
    @Column(nullable = false, unique = true)
    private String policyNumber;
    @Column(nullable = false)
    private String type;
    private double premium;
    private double coverage;
    private LocalDate startDate;
    private LocalDate endDate;
    private String status;

    @ManyToOne(optional = false)
    private Customer customer;

    @OneToMany(mappedBy = "policy")
    private List<Claim> claims = new ArrayList<>();

    public String getId() { return id; }
    public String getPolicyNumber() { return policyNumber; }
    public void setPolicyNumber(String policyNumber) { this.policyNumber = policyNumber; }
    public String getType() { return type; }
    public void setType(String type) { this.type = type; }
    public double getPremium() { return premium; }
    public void setPremium(double premium) { this.premium = premium; }
    public double getCoverage() { return coverage; }
    public void setCoverage(double coverage) { this.coverage = coverage; }
    public LocalDate getStartDate() { return startDate; }
    public void setStartDate(LocalDate startDate) { this.startDate = startDate; }
    public LocalDate getEndDate() { return endDate; }
    public void setEndDate(LocalDate endDate) { this.endDate = endDate; }
    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
    public Customer getCustomer() { return customer; }
    public void setCustomer(Customer customer) { this.customer = customer; }
    public List<Claim> getClaims() { return claims; }
}


